runtime/vm/extendedMessageNPE.cpp

```diff
@@ -22,10 +22,227 @@
+	UDATA j = 0;
+	UDATA arity = 0;
+	U_8 *string = J9UTF8_DATA(methodSig);
+	UDATA stringLength = J9UTF8_LENGTH(methodSig);
+	UDATA bufferSize = 0;
+	char *result = NULL;
+
+	PORT_ACCESS_FROM_VMC(vmThread);
+
+	/* first scan to calculate the buffer size required */
+	/* first character is '(' */
+	bufferSize += 1;
+	for (i = 1; ((')' != string[i]) && (i < stringLength)); i++) {
+		arity = 0;
+		while (('[' == string[i]) && (i < stringLength)) {
+			arity += 1;
+			i += 1;
+		}
+		switch (string[i]) {
+		case 'B': /* FALLTHROUGH */
+		case 'C': /* FALLTHROUGH */
+		case 'J':
+			/* byte, char, long */
+			bufferSize += 4;
+			break;
+		case 'D':
+			/* double */
+			bufferSize += 6;
+			break;
+		case 'F': /* FALLTHROUGH */
+		case 'S':
+			/* float, short */
+			bufferSize += 5;
+			break;
+		case 'I':
+			/* int */
+			bufferSize += 3;
+			break;
+		case 'L': {
+			i++;
+			UDATA objSize = 0;
+			while ((';' != string[i]) && (i < stringLength)) {
+				objSize++;
+				i++;
+			}
+			bufferSize += objSize;
+			break;
+		}
+		case 'Z':
+			/* boolean */
+			bufferSize += 7;
+			break;
+		default:
+			Trc_VM_ConvertMethodSignature_Malformed_Signature(vmThread, stringLength, string, i);
+			break;
+		}
+		bufferSize += (2 * arity);
+		if (')' != string[i + 1]) {
+			/* ", " */
+			bufferSize += 2;
+		}
+	}
+	/* ')' and the extra byte for '\0' */
+	bufferSize += 2;
+	Trc_VM_ConvertMethodSignature_Signature_BufferSize(vmThread, stringLength, string, bufferSize);
+	result = (char *)j9mem_allocate_memory(bufferSize, OMRMEM_CATEGORY_VM);
+	if (NULL != result) {
+		char *cursor = result;
+		UDATA availableSize = bufferSize;
+
+		memset(result, 0, bufferSize);
+		/* first character is '(' */
+		j9str_printf(PORTLIB, cursor, availableSize, "(");
+		cursor += 1;
+		availableSize -= 1;
+		for (i = 1; ((')' != string[i]) && (i < stringLength)); i++) {
+			arity = 0;
+			while (('[' == string[i]) && (i < stringLength)) {
+				arity += 1;
+				i += 1;
+			}
+			const char *elementType = NULL;
+			if ('L' == string[i]) {
+				i += 1;
+
+				UDATA objSize = 0;
+				while ((';' != string[i]) && (i < stringLength)) {
+					if ('/' == string[i]) {
+						*cursor = '.';
+					} else {
+						*cursor = string[i];
+					}
+					objSize += 1;
+					cursor += 1;
+					i += 1;
+				}
+				availableSize -= objSize;
+			} else {
+				switch (string[i]) {
+				case 'B':
+					elementType = "byte";
+					break;
+				case 'C':
+					elementType = "char";
+					break;
+				case 'D':
+					elementType = "double";
+					break;
+				case 'F':
+					elementType = "float";
+					break;
+				case 'I':
+					elementType = "int";
+					break;
+				case 'J':
+					elementType = "long";
+					break;
+				case 'S':
+					elementType = "short";
+					break;
+				case 'Z':
+					elementType = "boolean";
+					break;
+				default:
+					Trc_VM_ConvertMethodSignature_Malformed_Signature(vmThread, stringLength, string, i);
+					break;
+				}
+				UDATA elementLength = strlen(elementType);
```

Comment:
can you avoid the additional strlen?

Reply:
Not sure about the additional `strlen` reference. During earlier buffer size calculation, the `elementType` length was hardcoded without `strlen`.

Reply:
Resolving for now.